src/ert/callbacks.py

```diff
@@ -96,36 +96,47 @@ async def forward_model_ok(
     realization: int,
     iter: int,
     ensemble: Ensemble,
+    forward_model_ok_permanent_error_future: asyncio.Future[str] | None = None,
```

Comment:
how the Future is usually used is that when you do somewhere:
```python
await forward_model_ok_permanent_error_future
```

and somewhere else you get exception Ex you can propagate it to the future by 
```python
forward_model_ok_permanent_error_future.set_exception(Ex)
```

which then gets triggered by the line above (with await).